### PR TITLE
Add flake detector server

### PR DESCRIPTION
**Pull request by kargakis**
@stevekuznetsov @smarterclayton @deads2k 

https://prometheus-kube-system.svc.ci.openshift.org/graph?g0.range_input=1d&g0.expr=flake_percentage&g0.tab=0
https://flake-detector-experiment.svc.ci.openshift.org/prometheus

### CODE REVIEW

**Comment by stevekuznetsov** [comment]
Is this unmerged upstream, or are we cherrypicking it instead of rebasing our repo, or did you write it just now? Unclear what review you need

File: experiment/flake-detector/config.yaml [hunk 1]
```diff
@@ -0,0 +1,13 @@
+# Create with 'kubectl create cm config --from-file=config.yaml'
```
Comment (spxtr) [inline experiment/flake-detector/config.yaml]: Possible to instead just use info from `prow/config.yaml`?

File: experiment/flake-detector/main.go [hunk 2]
```diff
@@ -0,0 +1,220 @@
...
+	"log"
+	"net/http"
+	"sort"
+	"time"
+
+	"github.com/prometheus/client_golang/prometheus"
+	"github.com/prometheus/client_golang/prometheus/promhttp"
+)
+
+type prowJob struct {
```
Comment (spxtr) [inline experiment/flake-detector/main.go]: If we move this to `kubernetes/test-infra` then we can just use the info straight from prow.

File: experiment/flake-detector/config.yaml (hunk 1 above)
Comment (kargakis) [inline experiment/flake-detector/config.yaml]: Yeah, will switch to use it.

File: experiment/flake-detector/main.go (hunk 2 above)
Comment (kargakis) [inline experiment/flake-detector/main.go]: Agreed, I will try to open a PR later this week.

**Comment by kargakis** [comment]
Moved in https://github.com/kubernetes/test-infra/pull/4082

**Pull request closed without merging**